### PR TITLE
Feat input wrapper

### LINKED ISSUE #7682: Angular UI: Add wrapper components for form inputs

### PR DESCRIPTION
**Pull request by bariscanyilmaz**
### Description

Resolves #7682

Added wrapper components for checkbox and text input. 

### Checklist

- [x] I fully tested it as developer / designer and created unit / integration tests
- [x] I documented it (or no need to document or I will create a separate documentation issue)

### CODE REVIEW

File: npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      useExisting: forwardRef(() => CheckboxComponent),
+      multi: true,
+    },
+  ]
+})
+export class CheckboxComponent extends AbstractNgModelComponent {
+
+  @Input() label: string;
+  @Input() checkboxId!: string;
+  @Input() formControl!: string;
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts]: please remove

File: npm/ng-packs/packages/theme-shared/src/lib/theme-shared.module.ts [hunk 2]
```diff
@@ -38,7 +38,8 @@ import { CONFIRMATION_ICONS, DEFAULT_CONFIRMATION_ICONS } from './tokens/confirm
 import { PasswordComponent } from './components/password/password.component';
 import { CardModule } from './components/card/card.module';
 import { AbpVisibleDirective } from './directives';
-
+import { FormInputComponent } from './components/form-input/form-input.component';
+import {CheckboxComponent} from './components/checkbox/checkbox.component';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/theme-shared.module.ts]: please reformat

File: npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    await TestBed.configureTestingModule({
+      declarations: [CheckboxComponent],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(CheckboxComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.spec.ts]: ttest file is ok but basic test should be there.

File: npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    class="form-control"  
+    type="text"
+    [id]="inputId" 
+    [placeholder]="inputPlaceholder" 
+    [readonly]="inputReadonly"
+    [ngClass]="inputClass" 
+    [ngStyle]="inputStyle" 
+    (blur)="onBlur.next()"
+    (focus)="onFocus.next()" 
+    [(ngModel)]="value">
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.html]: I thinki change value throw sepereately. othervise user can not attach the valueChanges (change) that

**Review by mahmut-gundogdu — CHANGES_REQUESTED** [review]

File: npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  styleUrls: ['./checkbox.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CheckboxComponent),
+      multi: true,
+    },
+  ]
+})
+export class CheckboxComponent extends AbstractNgModelComponent {
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts]: component class names do not have a naming convention between FormInputComponent and CheckboxComponent. Please make FormCheckboxComponent or please make FormInputComponent to InputComponent

File: npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<div class="mb-3">
+  <label class="form-check-label" *ngIf="label" [ngClass]="labelClass" [for]="checkboxId" > {{label | abpLocalization}} </label>
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.html]: i think label should ordered after input

File: npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.html (hunk 4 above)
Comment (muhammedaltug) [reply npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.html]: AbstractNgModelComponent handles with valueChange

File: npm/ng-packs/packages/identity/src/lib/components/users/users.component.html [hunk 7]
```diff
@@ -55,12 +55,16 @@ <h3>{{ (selected?.id ? 'AbpIdentity::Edit' : 'AbpIdentity::NewUser') | abpLocali
               *ngFor="let roleGroup of roleGroups; let i = index; trackBy: trackByFn"
               class="form-check mb-2"
             >
-              <input
-                type="checkbox"
-                class="form-check-input"
-                [attr.id]="'roles-' + i"
+              <abp-checkbox 
+              *abpReplaceableTemplate="{
+                checkboxId:'roles-' + i,
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/identity/src/lib/components/users/users.component.html]: i think it should be
```html
<abp-checkbox 
    *abpReplaceableTemplate="{
        inputs: {
            checkboxId:'roles-' + I,
            formControl:roleGroup.controls[roles[i].name]
        },
        componentKey: inputKey,
```

File: npm/ng-packs/packages/identity/src/lib/components/users/users.component.html [hunk 8]
```diff
@@ -55,12 +55,16 @@ <h3>{{ (selected?.id ? 'AbpIdentity::Edit' : 'AbpIdentity::NewUser') | abpLocali
...
+                componentKey: inputKey,
+                formControl:roleGroup.controls[roles[i].name]
+              }" 
+                [checkboxId]="'roles-' + i"
                 [formControl]="roleGroup.controls[roles[i].name]"
-              />
+              >
+              </abp-checkbox>
               <label class="form-check-label" [attr.for]="'roles-' + i">{{ roles[i].name }}</label>
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/identity/src/lib/components/users/users.component.html]: you can pass label to checkbox

File: npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+      multi: true,
+    },
+  ]
+})
+export class FormCheckboxComponent extends AbstractNgModelComponent {
+
+  @Input() label?: string;
+  @Input() checkboxId!: string;
+  @Input() checkboxStyle = '';
+  @Input() checkboxClass = '';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts]: please give default class form-check-input

File: npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  @Input() checkboxReadonly = false;
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/checkbox/checkbox.component.ts:37]: plase add label class input

File: npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.ts [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+})
+export class FormInputComponent extends AbstractNgModelComponent {
+  @Input() inputId!: string;
+  @Input() inputReadonly: boolean = false;
+  @Input() label: string = '';
+  @Input() labelClass: string = '';
+  @Input() inputPlaceholder: string = '';
+  @Input() inputType: string = 'text';
+  @Input() inputStyle: string = '';
+  @Input() inputClass: string = '';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.ts]: please give default form-control

File: npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.ts [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+      useExisting: forwardRef(() => FormInputComponent),
+      multi: true,
+    },
+  ]
+})
+export class FormInputComponent extends AbstractNgModelComponent {
+  @Input() inputId!: string;
+  @Input() inputReadonly: boolean = false;
+  @Input() label: string = '';
+  @Input() labelClass: string = '';
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/form-input/form-input.component.ts]: please give default form-label

**Review by muhammedaltug — CHANGES_REQUESTED** [review]

**Review by muhammedaltug — APPROVED** [review]

**Review by mahmut-gundogdu — APPROVED** [review]

**Pull request merged**